src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -218,6 +223,37 @@ private void ComputeIfModifiedSince()
             }
         }
 
+        private void ComputeIfRange(RequestHeaders requestHeaders, DateTimeOffset? lastModified = null, EntityTagHeaderValue etag = null)
+        {
+            // 14.27 If-Range
+            var ifRangeHeader = requestHeaders.IfRange;
+            if (ifRangeHeader != null)
+            {
+                // If the validator given in the If-Range header field matches the
+                // current validator for the selected representation of the target
+                // resource, then the server SHOULD process the Range header field as
+                // requested.  If the validator does not match, the server MUST ignore
+                // the Range header field.
+                bool ignoreRangeHeader = false;
+                if (ifRangeHeader.LastModified.HasValue)
+                {
+                    if (lastModified.HasValue && lastModified > ifRangeHeader.LastModified)
+                    {
+                        ignoreRangeHeader = true;
+                    }
+                }
+                else if (etag != null && ifRangeHeader.EntityTag != null && !ifRangeHeader.EntityTag.Compare(etag, useStrongComparison: true))
+                {
+                    ignoreRangeHeader = true;
+                }
+
+                if (ignoreRangeHeader)
+                {
+                    _ifRangeState = PreconditionState.IgnoreRangeRequest;
```

Comment:
I don't think we need this new state. The other states all result in specific status codes, but this one causes the same behavior as ShouldProcess. How about just setting _isRangeRequest = false;?
I guess you'd have to reverse the call order of ComputeIfRange and ParseRange.